app/Http/Controllers/ScoresController.php

```diff
@@ -54,6 +61,18 @@ public function download($rulesetOrSoloId, $id = null)
             'Content-Type' => 'application/x-osu-replay',
         ];
 
+
+        if (Auth::user()->user_id !== $score->user->user_id) {
+            $score->user->statistics($ruleset, true)->increment('replay_popularity');
+
+            $currentMonth = UserCountryHistory::formatDate(CountryChangeTarget::currentMonth());
+            $score->user->replaysWatchedCounts->where('year_month', $currentMonth)->first()->increment('count');
+
+            if ($score instanceof ScoreBest) {
+                $score->replayViewCount()->increment('play_count');
+            }
```

Comment:
the relevant rows need to be created first if not exist yet?

Reply:
forgive me for not knowing much about user flow, but shouldn't they always exist? if not, which ones might not so I can account for it?

Reply:
I think only `statistics` is guaranteed to exist